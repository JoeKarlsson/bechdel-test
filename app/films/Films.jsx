/* jshint esversion: 6 */
'use strict';

import React from 'react';
import { Link } from 'react-router';
import FilmList from './FilmList.jsx';
import styles from './Films.scss';
import * as $ from'jquery';

const Films = React.createClass({
  getInitialState() {
    return {
      films: []
    }
  },

  getAllFilms: function() {
    $.ajax({
      url: "/api/film",
      method: 'GET',
      dataType: 'json',
      cache: false,
      success: function(data) {
        console.log(data, 'data')
        this.setState({ films: data });
      }.bind(this),
      error: function(xhr, status, err) {
        console.error(this.props, status, err.toString());
      }.bind(this)
    });
  },

  componentDidMount: function() {
    this.getAllFilms();
  },

  render: function() {
    return (
      <div className={styles.filmList}>
        <div className='u-full-width'>
          <h1>Films</h1>
          <Link to={'/film/new'}><button>Add Film</button></Link>
        </div>
        <div className='row'>
          <FilmList films={this.state.films} />
        </div>
      </div>
    )
  }
});

export default Films;