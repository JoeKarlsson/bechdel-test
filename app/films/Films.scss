$tomato: tomato;

.filmList {
  background-color: $tomato;
}