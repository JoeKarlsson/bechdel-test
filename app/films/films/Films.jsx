import React from 'react';
import { Link } from 'react-router';
import FilmList from './FilmList.jsx';
import skeleton from '../../shared/css/skeleton.css';
import styles from './Films.scss';
import * as $ from'jquery';

class Films extends React.Component {
  constructor() {
    super();
    this.state = {
      films: [],
    };
 }

  getAllFilms() {
    $.ajax({
      url: '/api/film',
      method: 'GET',
      dataType: 'json',
      cache: false,
      success: function(data) {
        this.setState({ films: data });
      }.bind(this),
      error: function(xhr, status, err) {
        console.error(this.props, status, err.toString());
      }.bind(this)
    });
  }

  componentDidMount() {
    this.getAllFilms();
  }

  render() {
    return (
      <div className={styles.films}>
        <div className={skeleton.u_full_width}>
          <div className='row'>
            <h1>Bechdel Test Visualizer</h1>
            <p>
              The Bechdel Test Visualizer was a collaborative project between a English Scholar and a Software Engineer. With the tool we’ve created, the process of determing whether a film passes the Beschdel Test is automated, which allows massive amounts of data to be generated with ease. Thus, data can be produced for large bodies of film, i.e. a certain director’s filmography, a certain actress’ body of work, or for the films released in a specific year.
            </p>
          </div>
        </div>
        <h3>Films</h3>
        <Link to={'/film/new'}>
          <button>Add a Film</button>
        </Link>
        <div className='row'>
          <FilmList films={this.state.films} />
        </div>
        <p>
          If you have any suggestions for how to grow the application, we would love to hear from <a target="blank" href="https://www.callmejoe.net/contact/">you</a>.
        </p>
      </div>
    )
  }
}

Films.propTypes = {
  films: React.PropTypes.array,
};

Films.defaultProps = {
  films: [],
};

export default Films;
