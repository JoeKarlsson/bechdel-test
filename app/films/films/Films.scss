.films_Header {
  display: flex;
  justify-content: space-between;
}