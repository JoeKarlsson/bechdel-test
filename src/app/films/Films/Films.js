import React, { Component } from 'react';
import Hero from './Hero/Hero';
import ErrorBoundary from '../../shared/ErrorBoundary/ErrorBoundary';
import FilmList from './FilmList/FilmList';
import api from '../../helper/api';
// import './Films.scss';

class Films extends Component {
	static renderLoading() {
		return <div>Loading...</div>;
	}

	static renderError() {
		return <div>Im sorry! Please try again.</div>;
	}

	constructor() {
		super();
		this.state = {
			films: [],
			loading: false,
		};
		this.getAllFilms = this.getAllFilms.bind(this);
	}

	componentDidMount() {
		this.getAllFilms();
	}

	getAllFilms() {
		const url = '/api/filmasdas/';
		const options = {
			method: 'GET',
		};

		this.setState({
			loading: true,
		});

		api(url, options).then(data => {
			this.setState({
				films: data,
				loading: false,
			}).catch(() => {
				this.setState({
					loading: false,
				});
			});
		});
	}

	renderFilms() {
		return (
			<div className="films">
				<Hero />
				<div className="row">
					<ErrorBoundary>
						<FilmList films={this.state.films} />
					</ErrorBoundary>
				</div>
			</div>
		);
	}

	render() {
		if (this.state.loading) {
			return Films.renderLoading();
		} else if (this.state.films) {
			return Films.renderFilms();
		}
		return this.renderError();
	}
}

Films.propTypes = {};

Films.defaultProps = {};

export default Films;
