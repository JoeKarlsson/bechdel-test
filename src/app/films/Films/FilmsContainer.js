import React, { Component } from 'react';
import Films from './Films';
import api from '../../helper/api';

class FilmsContainer extends Component {
	constructor() {
		super();
		this.state = {
			films: [],
			loading: false,
		};
		this.getAllFilms = this.getAllFilms.bind(this);
	}

	componentDidMount() {
		this.getAllFilms();
	}

	getAllFilms() {
		const url = '/api/film';
		const options = {
			method: 'GET',
		};

		this.setState({
			loading: true,
		});

		api(url, options)
			.then(data => {
				this.setState({
					films: data,
					loading: false,
				});
			})
			.catch(err => {
				console.err(err);
				this.setState({
					loading: false,
				});
			});
	}

	render() {
		return <Films {...this.state} />;
	}
}

FilmsContainer.propTypes = {};

FilmsContainer.defaultProps = {};

export default FilmsContainer;
