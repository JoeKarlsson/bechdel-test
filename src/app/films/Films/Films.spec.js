import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Films from './Films';

jest.mock('../../helper/api');

configure({ adapter: new Adapter() });

describe('Films', () => {
	let wrapper;
	let inst;

	beforeEach(() => {
		wrapper = shallow(
			<MemoryRouter>
				<Films />
			</MemoryRouter>
		);
		inst = wrapper.instance();
	});

	describe('rendering', () => {
		describe('initial state', () => {
			it('match the snapshot', () => {
				const component = renderer.create(
					<MemoryRouter>
						<Films />
					</MemoryRouter>
				);
				const tree = component.toJSON();
				expect(tree).toMatchSnapshot();
			});
			it('is rendered correctly', () => {
				expect(wrapper).toHaveLength(1);
			});
			it('should not have any inital props', () => {
				const initialProps = inst.props;
				const expectedProps = {
					children: <Films />,
				};
				expect(initialProps).toMatchObject(expectedProps);
			});
			it('should not have any inital state', () => {
				const initialState = inst.state;
				const expectedState = {};
				expect(initialState).toMatchObject(expectedState);
			});
		});
	});
});
