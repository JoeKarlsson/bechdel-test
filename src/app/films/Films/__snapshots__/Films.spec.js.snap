// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Films rendering initial state match the snapshot 1`] = `
<div
  className="films"
>
  <div
    className="u_full_width"
  >
    <div
      className="row"
    >
      <h1>
        Bechdel.io
      </h1>
      <p>
        The Bechdel.io was a collaborative project between a English Scholar and a Software Engineer. With the tool we’ve created, the process of determing whether a film passes the Beschdel Test is automated, which allows massive amounts of data to be generated with ease. Thus, data can be produced for large bodies of film, i.e. a certain director’s filmography, a certain actress’ body of work, or for the films released in a specific year.
      </p>
    </div>
  </div>
  <div
    className="films_Header"
  >
    <h3>
      films
    </h3>
    <a
      aria-current={false}
      className={undefined}
      href="/film/new"
      onClick={[Function]}
      style={undefined}
    >
      <button>
        Add a Film
      </button>
    </a>
  </div>
  <div
    className="row"
  >
    <div
      className="FilmList"
    />
  </div>
  <p>
    If you have any suggestions for how to grow the application, we would love to hear from
    <a
      href="https://www.callmejoe.net/contact/"
      target="blank"
    >
       
      you
    </a>
    .
  </p>
</div>
`;
